src/Tribe/Integrations/Freemius.php

```diff
@@ -0,0 +1,386 @@
+		$settings_url = $this->get_settings_url();
+
+		if ( false !== strpos( $connect_url, 'fs_action' ) ) {
+			$action = $this->slug . '_reconnect';
+
+			$settings_url = add_query_arg( [
+				'nonce'     => wp_create_nonce( $action ),
+				'fs_action' => $action,
+			], $settings_url );
+		}
+
+		return $settings_url;
+	}
+
+	/**
+	 * Get the Settings page URL.
+	 *
+	 * @since TBD
+	 *
+	 * @return string The Settings page URL.
+	 */
+	public function get_settings_url() {
+		return admin_url( $this->get_settings_path() );
+	}
+
+	/**
+	 * Get the plugin icon URL.
+	 *
+	 * @since TBD
+	 *
+	 * @return string The plugin icon URL.
+	 */
+	public function get_plugin_icon_url() {
+		$class = $this->object_class;
+
+		return $class::instance()->plugin_url . '/src/resources/images/' . $this->slug . '.svg';
+	}
+
+	/**
+	 * Get the Settings page path.
+	 *
+	 * @since TBD
+	 *
+	 * @return string The Settings page path.
+	 */
+	public function get_settings_path() {
+		if ( class_exists( 'Tribe__Events__Main' ) ) {
+			$url = sprintf( 'edit.php?post_type=%s&page=%s', Tribe__Events__Main::POSTTYPE, $this->page );
+		} else {
+			$url = sprintf( 'admin.php?page=%s', $this->page );
+		}
+
+		return $url;
+	}
+
+	/**
+	 * Action to skip activation since Freemius code does not skip correctly here.
+	 *
+	 * @since  TBD
+	 *
+	 * @return bool Whether activation was skipped.
+	 */
+	public function action_skip_activation() {
+		$fs_action = tribe_get_request_var( 'fs_action' );
+
+		// Prevent fatal errors.
+		if ( ! function_exists( 'fs_redirect' ) || ! function_exists( 'fs_is_network_admin' ) ) {
+			return false;
+		}
+
+		// Actually do the skipping of connection, since Freemius code does not do this.
+		if ( $this->slug . '_skip_activation' !== $fs_action ) {
+			return false;
+		}
+
+		check_admin_referer( $this->slug . '_skip_activation' );
+
+		$this->instance->skip_connection( null, fs_is_network_admin() );
+
+		fs_redirect( $this->instance->get_after_activation_url( 'after_skip_url' ) );
+
+		return true;
+	}
+
+	/**
+	 * Filter the content for the Freemius Popup.
+	 *
+	 * @since  TBD
+	 *
+	 * @param string $message         The message content.
+	 * @param string $user_first_name The first name of user.
+	 * @param string $product_title   The product title.
+	 * @param string $user_login      The user_login of user.
+	 * @param string $site_link       The site URL.
+	 * @param string $freemius_link   The Freemius URL.
+	 *
+	 * @return string
+	 */
+	public function filter_connect_message_on_update(
+		$message, $user_first_name, $product_title, $user_login, $site_link, $freemius_link
+	) {
+		$class = $this->object_class;
+
+		wp_enqueue_style( 'tribe-' . $this->slug . '-freemius', $class::instance()->plugin_url . '/src/resources/css/freemius.css' );
+
+		// Add the heading HTML.
+		$plugin_name = $this->name;
+		$title       = '<h3>' . sprintf( esc_html__( 'We hope you love %1$s', 'event-tickets' ), $plugin_name ) . '</h3>';
+		$html        = '';
```

Comment:
:no_entry_sign: A gettext call containing placeholders was found, but was not accompanied by a "translators:" comment on the line above to clarify the meaning of the placeholders.

<sub>hash: db461ef1a73626cb824fa73be0f1304d</sub>